src/viking.js

```diff
@@ -1,15 +1,77 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength){
+    this.health = health;
+    this.strength = strength;
+  }
+  attack() {
+    return this.strength;
+  }
+  receiveDamage(damage){
+    this.health -= damage;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength){
+    super(health, strength);
+    this.name = name;
+  }
+
+  receiveDamage(damage){
+    this.health -= damage;
+    if(this.health > 0) {
+      let message = `${this.name} has received ${damage} points of damage`
+      return message;
+    }else {
+      let message = `${this.name} has died in act of combat`
+      return message;
+    }
+  }
+
+  battleCry() {
+    return 'Odin Owns You All!'
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  receiveDamage(damage){
+    this.health -= damage;
+    if(this.health > 0) {
+      let message = `A Saxon has received ${damage} points of damage`
+      return message;
+    }else {
+      let message = `A Saxon has died in combat`
+      return message;
+    }
+  }
+}
 
 // War
-class War {}
+class War {
+  constructor() {
+    this.vikingArmy = [];
+    this.saxonArmy = []; 
+  }
+
+  addViking(viking){
+    this.vikingArmy.push(viking)
+  }
 
+  addSaxon(saxon){
+    this.saxonArmy.push(saxon)
+  }
+
```

Comment:
Les agrego la guerra por que #spartans 
`
class War {
  constructor() {
    this.vikingArmy = [];
    this.saxonArmy = [];
  }

  addViking(aVikingObject) {
    this.vikingArmy.push(aVikingObject);
  }

  addSaxon(aSaxonObject) {
    this.saxonArmy.push(aSaxonObject);
  }

  vikingAttack() {
    let randomVikingIndex = Math.floor(Math.random() * this.vikingArmy.length);
    let randomViking = this.vikingArmy[randomVikingIndex];
    let randomSaxonIndex = Math.floor(Math.random() * this.saxonArmy.length);
    let randomSaxon = this.saxonArmy[randomSaxonIndex];
    let thingToReturnAtTheEnd = randomSaxon.receiveDamage(
      randomViking.attack()
    );
    if (randomSaxon.health <= 0) {
      this.saxonArmy.splice(randomSaxonIndex, 1);
    }

    return thingToReturnAtTheEnd;
  }

  saxonAttack() {
    let randomVikingIndex = Math.floor(Math.random() * this.vikingArmy.length);
    let randomViking = this.vikingArmy[randomVikingIndex];
    let randomSaxonIndex = Math.floor(Math.random() * this.saxonArmy.length);
    let randomSaxon = this.saxonArmy[randomSaxonIndex];

    let thingToReturnAtTheEnd = randomViking.receiveDamage(
      randomSaxon.attack()
    );
    if (randomViking.health <= 0) {
      this.vikingArmy.splice(randomVikingIndex, 1);
    }

    return thingToReturnAtTheEnd;
  }

  showStatus() {
    if (this.vikingArmy.length > 0 && this.saxonArmy.length > 0) {
      return 'Vikings and Saxons are still in the thick of battle.';
    } else if (this.vikingArmy.length > 0) {
      return 'Vikings have won the war of the century!';
    } else {
      return 'Saxons have fought for their lives and survived another day...';
    }
  }
}
`